assets/js/graphql/get-traits.js

```diff
@@ -0,0 +1,160 @@
+              pubMedId
+              doi
+            }
+          }
+        }
+        gwas {
+          identifier
+          synopsis
+          description
+          genotypes
+          organism {
+            genus
+            species
+          }
+          dataSet {
+            publication {
+              firstAuthor
+              pubMedId
+              doi
+            }
+          }
+        }
+      }
+      pageInfo {
+        currentPage
+        pageSize
+        numResults
+        pageCount
+        hasPreviousPage
+        hasNextPage
+      }
+    }
+  }
+`;
+
+
+/**
+ * Gets traits from GraphQL.
+ * @param {object} queryData - An object containing zero or more variables for the GraphQL query.
+ * @param {object} pageData - An object containing pagination data for the GraphQL query, if any.
+ * @param {object} options - An object containing optional parameters for the HTTP request,
+ * namely, an optional `AbortSignal` instance that can be used to cancel the request mid-flight.
+ * @returns {Promise} A `Promise` that resolves to the result of the GraphQL query.
+ */
+export function getTraits(queryData={}, pageData={}, options={}) {
+  const genus = queryData.genus;
+  const species = queryData.species;
+  const studyType = queryData.type;
+  const name = queryData.traits;
+  const publicationId = queryData.pubId;
+  const author = queryData.author;
+  if (studyType === 'QTL') {
+    studyType = 'QTLStudy';
+  }
+  const {page, pageSize} = pageData;
+  const variables = {
+    genus,
+    species,
+    studyType,
+    name,
+    publicationId,
+    author,
+    page,
+    pageSize: 10,
+  };
+  const {abortSignal} = options;
+  return query(getTraitsQuery, variables, abortSignal);
+}
+
+/**
+ * Converts GraphQL `TraitsResults` into the `PaginatedSearchResults<TraitSearchResult[]>` used by the
+ * `LisTraitSearchElement` (`<lis-trait-search-element>`) Web Component.
+ * @param {object} data - An object containing the data portional of the GraphQL query HTTP response.
+ * @returns {object} A `PaginatedSearchResults<TraitSearchResult[]>` object.
+ */
+export function traitsDataToSearchResults(data) {
+  // extract the page info
+  const {hasNextPage: hasNext, numResults, pageSize, pageCount: numPages} = data.traits.pageInfo;
+  // flatten results
+  const results = data.traits.results.map(({name, ...trait}) => {
+    const type = trait.qtlStudy ? 'QTL' : 'GWAS';
+    const study = trait.qtlStudy || trait.gwas;
+    const {identifier, synopsis, description, genotypes} = study;
+    return {
+      identifier,
+      type,
+      synopsis,
+      description,
+      name,
+      genotypes
+    };
+  });
+  // return the results
+  return {hasNext, numResults, pageSize, numPages, results};
+}
+
+/**
+ * The trait search function to use for the `searchFunction` property of the `LisTraitAssociationSearchElement`
+ * (`<lis-trait-association-search-element>`) Web Component.
+ * @param {string} query - The search query.
```

Comment:
`query`  should be `queryData`, and these parameter descriptions are virtually no more descriptive than the parameter names themselves. Please copy the corresponding descriptions from [`get-genes.js`](https://github.com/legumeinfo/jekyll-theme-legumeinfo/blob/main/assets/js/graphql/get-genes.js#L98).